subsys/bluetooth/host/keys.h

```diff
@@ -96,14 +96,26 @@ enum {
 
 struct bt_keys_link_key {
 	bt_addr_t		addr;
+	u8_t                    storage_start[0];
 	u8_t                    flags;
 	u8_t			val[16];
 };
+#define BT_KEYS_LINK_KEY_STORAGE_LEN     (sizeof(struct bt_keys_link_key) - \
+	offsetof(struct bt_keys_link_key, storage_start))
 
 struct bt_keys_link_key *bt_keys_get_link_key(const bt_addr_t *addr);
 struct bt_keys_link_key *bt_keys_find_link_key(const bt_addr_t *addr);
 void bt_keys_link_key_clear(struct bt_keys_link_key *link_key);
 void bt_keys_link_key_clear_addr(const bt_addr_t *addr);
+#if defined(CONFIG_BT_SETTINGS)
+void bt_keys_link_key_store(struct bt_keys_link_key *link_key);
```

Comment:
Removing the `#if defined(CONFIG_BT_SETTINGS)` around the definition and implementation and using `if (IS_ENABLED(CONFIG_BT_SETTINGS))` is the preferred approach. (I know it is the same way for `bt_keys_store` but we should change that).